pkg/didcomm/common/service/service.go

```diff
@@ -122,3 +132,22 @@ func (m *DIDCommMsg) ThreadID() (string, error) {
 
 	return "", ErrThreadIDNotFound
 }
+
+// MessageHeader is header of the generic message
+type MessageHeader struct {
```

Comment:
Do we need a separate struct? @fqutishat @sudeshrshetty

Reply:
@troyronda DIDComm Header may not have all fields of interest.

Either we pass whole DIDComm Msg or Portion of it as a struct

Reply:
As discussed, moved `~purpose` to header and Accept() criteria will use DID Comm msg header.